projects/core/directives/dropdown/dropdown.component.ts

```diff
@@ -0,0 +1,120 @@
+import {AnimationOptions} from '@angular/animations';
+import {
+    ChangeDetectionStrategy,
+    Component,
+    ElementRef,
+    HostBinding,
+    Inject,
+    Optional,
+} from '@angular/core';
+import {WINDOW} from '@ng-web-apis/common';
+import {TuiDestroyService, tuiGetClosestFocusable, tuiPx} from '@taiga-ui/cdk';
+import {TuiRectAccessor} from '@taiga-ui/core/abstract';
+import {tuiDropdownAnimation} from '@taiga-ui/core/animations';
+import {TuiDropdownAnimation} from '@taiga-ui/core/enums';
+import {TuiPositionService} from '@taiga-ui/core/services';
+import {TUI_ANIMATION_OPTIONS} from '@taiga-ui/core/tokens';
+import {TuiPoint} from '@taiga-ui/core/types';
+import {Observable} from 'rxjs';
+import {takeUntil} from 'rxjs/operators';
+
+import {TuiDropdownDirective} from './dropdown.directive';
+import {TuiDropdownHoverDirective} from './dropdown-hover.directive';
+import {TUI_DROPDOWN_OPTIONS, TuiDropdownOptions} from './dropdown-options.directive';
+
+/**
+ *  This component is used to show template in a portal using default style of white rounded box with a shadow
+ */
+// @bad TODO: OnPush
+@Component({
+    selector: `tui-dropdown`,
+    templateUrl: `./dropdown.template.html`,
+    styleUrls: [`./dropdown.style.less`],
+    changeDetection: ChangeDetectionStrategy.Default,
```

Comment:
How can we improve this component with onpush in the future?

Reply:
There's a TODO for it.